### ISSUE TITLE
endian-ness warning

### ISSUE THREAD

**Issue by richli** [Unidata/netcdf4-python#364]
I noticed a change with the behavior of netcdf4-python, starting with 1.1.5. When I open a netcdf4 file, the following warning is raised:

```python
UserWarning: endian-ness of dtype and endian kwarg do not match, using endian kwarg
```

Here's a minimal example in IPython that triggers the warning:
```python
In [1]: from netCDF4 import Dataset

In [2]: f = Dataset("test.nc", "w", format='NETCDF4')

In [3]: f.createDimension("record", 10)
Out[3]: <class 'netCDF4.Dimension'>: name = 'record', size = 10

In [4]: f.createVariable("x", "i4", ("record", ))
Out[4]: 
<class 'netCDF4.Variable'>
int32 x(record)
unlimited dimensions: 
current shape = (10,)
filling on, default _FillValue of -2147483647 used

In [5]: f.close()

In [6]: f = Dataset("test.nc", "r")
/usr/bin/ipython:1: UserWarning: endian-ness of dtype and endian kwarg do not match, using endian kwarg
  #!/usr/bin/python3

In [7]: f.close()
```

Here's the output from `ncdump`. The variable is little-endian.
```bash
$ ncdump -h -s test.nc
netcdf test.nc {
dimensions:
	record = 10 ;
variables:
	int x(record) ;
		x:_Storage = "contiguous" ;
		x:_Endianness = "little" ;

// global attributes:
		:_Format = "netCDF-4" ;
}
```

Is there something I should be doing differently to not trigger the warning or is this a regression?

**Comment by jswhit** [comment]
I think this is a bug - you should not be getting that warning.

**Comment by jswhit** [comment]
fix in pull request #365

**Comment by jswhit** [comment]
merged pull request #365, closing.  Thanks for the report.
